compiler/rustc_resolve/src/imports.rs

```diff
@@ -866,6 +868,8 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                 (source, target, bindings, type_ns_only)
             }
             ImportKind::Glob { .. } => {
+                // FIXME: Use mutable resolver directly as a hack, this should be an output of
+                // specualtive resolution.
```

Comment:
It's better to use `SmartResolver` in `resolve_import` signature and use something like `res_mut_unchecked` in places where resolver is currently mutated, but should not be.